Add new FerretDB v0.9.3 release blogpost (#2201)
FerretDB/FerretDB: website/blog/2023-03-14-ferretdb-v-0-9-2.md

---
title: "FerretDB v.0.9.3 - Improved Aggregation Pipeline Support"
slug: ferretdb-v-0-9-3-improved-aggregation-pipeline-support
authors: [alex]
description: >
  We are thrilled to announce the release of FerretDB v.0.9.3, which includes exciting new features and improvements, and we can't wait for you to try it out
image: /img/blog/ferretdb_v0.9.3.jpg
tags: [release]
---

![FerretDB v0.9.3](/img/blog/ferretdb_v0.9.3.jpg)

> We are thrilled to announce the release of [FerretDB v.0.9.3](https://github.com/FerretDB/FerretDB/releases/tag/v0.9.3), which includes exciting new features and improvements, and we can't wait for you to try it out.

Reviewer: ```suggestion
We are thrilled to announce the release of [FerretDB v0.9.3](https://github.com/FerretDB/FerretDB/releases/tag/v0.9.3), which includes exciting new features and improvements, and we can't wait for you to try it out.
```

Let's standardize with the release tag.

<!--truncate-->

> This new release is quite monumental for us at FerretDB](https://www.ferretdb.io/), because it brings us one step closer to the release of FerretDB GA, which promises a truly standard open-source document database with MongoDB compatibility, as well as backend engines for [PostgreSQL](https://www.postgresql.org/), [Tigris](https://www.tigrisdata.com/), and potentially many others.

Reviewer 2: ```suggestion
This new release is quite monumental for us at [FerretDB](https://www.ferretdb.io/), because it brings us one step closer to the release of FerretDB GA, which promises a truly standard open-source document database with MongoDB compatibility, as well as backend engines for [PostgreSQL](https://www.postgresql.org/), [Tigris](https://www.tigrisdata.com/), and potentially many others.
```

> FerretDB v.0.9.3 builds on our previous release and promises better performance and database experience for FerretDB with enhanced support for new aggregation pipeline stages and more pushdown queries.

Reviewer: ```suggestion
FerretDB v0.9.3 builds on our previous release and promises better performance and database experience for FerretDB with enhanced support for new aggregation pipeline stages and more pushdown queries.
```
same here

    order_total: 10.99
}
```

We can use the `$sort` aggregation pipeline stage to sort the documents in ascending or descending order based on a specific field.
For example, to sort the documents in descending order based on the `order_total` field, set the value to `-1`:

```js
db.orders.aggregate([
    { $sort: { order_total: -1 } }
])
```

This will return the documents, sorted in descending order based on the `order_total` field.

You can use the `$group` aggregation stage to group documents together.
For example, the following query groups and counts the total number of documents in the `orders` collection.

```js
db.orders.aggregate([
    {
        $group: {
            _id: null,
            count: { $count: {} }
        }
    }
])
```

As part of our mission to effectively enable more backends to run MongoDB workloads, basic aggregation pipeline support is now available for [Tigris](https://www.tigrisdata.com/).

## Bug Fixes

FerretDB v.0.9.3 has had several enhancements and bug fixes since the last release, indicating our commitment to building a reliable document database for our users.

We’ve fixed dot notation errors for `$max`, `$min`, and `$mul` update operators.

> Previously, there were causing issues by failing to return errors when attempting to update paths that were not valid.

Reviewer: ```suggestion
Previously, they were causing issues by failing to return errors when attempting to update paths that were not valid.
```

The fixes now ensure that update operation will return an error when attempting to update an invalid path.

We’ve also addressed issues with querying an embedded field in an array, especially with the `$pullAll` operator, which was preventing it from removing all instances from an array.

> One of our contributors [@yu-re-ka](https://github.com/yu-re-ka) helped fixed `saslStart` for particular clients, making it possible to achieve compatibility with [FastNetMon](https://fastnetmon.com/).

Reviewer 2: We don't actually know yet if compatibility was achieved. It definitely did not work before, but it might also not work now. So it was _improved_, but not achieved

Author: PTAL
